drivers/usb/device/usb_dc_stm32.c

```diff
@@ -846,8 +846,13 @@ int usb_dc_ep_mps(const u8_t ep)
 
 int usb_dc_detach(void)
 {
-	LOG_ERR("Not implemented");
-
+	// stop usb dev
+	HAL_PCD_Stop(&usb_dc_stm32_state.pcd);
+	// take care that the endpoints are closed (usb_dc_ep_disable)
+	HAL_PCD_EP_Close(&usb_dc_stm32_state.pcd, EP0_IN);
+	HAL_PCD_EP_Close(&usb_dc_stm32_state.pcd, EP0_OUT);
```

Comment:
is it strictly necessary? what about the other endpoints?

Reply:
As far as I have seen there are only EP0_IN and EP0_OUT in that code. Guess stopping the usb, closing the endpoints and deinit is the most explicit and clean ending.